org/lateralgm/components/CodeTextArea.java

```diff
@@ -97,8 +98,8 @@ public String getString(String key, String def)
 
 	protected static Timer timer;
 	protected Integer lastUpdateTaskID = 0;
-	private Set<SortedSet<String>> resourceKeywords = new HashSet<SortedSet<String>>();
-	protected Completion[] completions;
+	private static Set<SortedSet<String>> resourceKeywords = new HashSet<SortedSet<String>>();
```

Comment:
This isn't thread-safe and precludes the possibility of LGM having multiple projects open. But I don't think we're using threads with this and LGM will clearly never support that, anyway.

Reply:
Exactly, that's what I mentioned above. Even if LGM does later become multiproject, it will want a keyword context for each project, not for each individual code text area (otherwise it'll be really wasteful).